network/unix_domain_server.cc

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright 2025 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "unix_domain_server.h"
+#include "log.h"
+
+namespace Network
+{
+
+DefaultMessageHandler::DefaultMessageHandler() {}
+
+void DefaultMessageHandler::OnConnect()
+{
+    LOGI("DefaultMessageHandler::OnConnect()");
+}
+
+void DefaultMessageHandler::HandleMessage(std::unique_ptr<ISerializable> message,
+                                          SocketConnection*              client_conn)
+{
+    if (!message)
+    {
+        LOGI("DefaultMessageHandler::HandleMessage: Null message.");
+        return;
+    }
+    if (!client_conn)
+    {
+        LOGI("DefaultMessageHandler::HandleMessage: Null client connection.");
+        return;
+    }
+
+    switch (static_cast<MessageType>(message->GetMessageType()))
```

Comment:
is there a reason why we want to return a uint32_t from GetMessageType(), can we return MessageType directly to avoid the casting?